astropy/coordinates/sky_coordinate.py

```diff
@@ -1225,6 +1225,92 @@ def from_pixel(cls, xp, yp, wcs, origin=0, mode='all'):
         """
         return pixel_to_skycoord(xp, yp, wcs=wcs, origin=origin, mode=mode, cls=cls)
 
+    def radial_velocity_correction(self, kind='barycentric', obstime=None,
+                                   location=None):
+        """
+        Compute the correction required to convert a radial velocity at a given
+        time and place to a barycentric or heliocentric velocity.
+
+        Parameters
+        ----------
+        kind : str
+            The kind of velocity correction.  Must be 'barycentric' or
+            'heliocentric'.
+        obstime : `~astropy.time.Time` or None, optional
+            The time at which to compute the correction.  If `None`, the
+            ``obstime`` frame attribute on the `SkyCoord` will be used. (Raises
+            an exception if the `SkyCoord` and this argument are in conflict.)
+        location : `~astropy.coordinates.EarthLocation` or None, optional
+            The observer location at which to compute the correction.  If
+            `None`, the  ``location`` frame attribute on the `SkyCoord` will be
+            used. (Raises an exception if the `SkyCoord` and this argument are
+            in conflict)
+
+
+        Returns
+        -------
+        vcorr : astropy.units.Quantity with velocity units
+            The  correction with a positive sign.  I.e., *add* this
+            to an observed radial velocity to get the heliocentric velocity.
+
+        Notes
+        -----
+        The algorithm here is sufficient to perform corrections at the ~1 to
+        10 m/s level, but has not been validated at higher prevision.  Future
```

Comment:
Typo: prevision -> precision. Although I personally sometimes get confused by "higher precision" because it's really a lower error. Maybe "better precision"?